From 6252f34f81cc10a8445362bb00dafa8d3f5e3494 Mon Sep 17 00:00:00 2001
From: Jiang Xin <zhiyou.jx@alibaba-inc.com>
Date: Tue, 28 Jun 2022 19:08:29 +0800
Subject: [PATCH 10/13] black-hole: implementation for builtin_smart_vtable

"builtin_smart_vtable" is the implementation for builtin protocols,
such as SSH and file protocol. Pass the "black-hole" option from
the transport layer to builtin_smart_vtable, and will pass option
to "git-fetch-pack".

Signed-off-by: Jiang Xin <zhiyou.jx@alibaba-inc.com>
---
 transport.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/transport.c b/transport.c
index 4e4b0b9aad..f1f33bc11b 100644
--- a/transport.c
+++ b/transport.c
@@ -374,6 +374,7 @@ static int fetch_refs_via_pack(struct transport *transport,
 	struct ref *refs_tmp = NULL;
 
 	memset(&args, 0, sizeof(args));
+	args.black_hole = data->options.black_hole;
 	args.uploadpack = data->options.uploadpack;
 	args.keep_pack = data->options.keep;
 	args.lock_pack = 1;
-- 
patchwork
